src/main/scala/nodes/learning/GaussianMixtureModel.scala

```diff
@@ -16,11 +17,16 @@ import workflow.{Transformer, Estimator}
  * @param variances Cluster variances (diagonal)
  * @param weights Cluster weights.
  */
-class GaussianMixtureModel(
-    val means: DenseMatrix[Double],
-    val variances: DenseMatrix[Double],
-    val weights: DenseVector[Double])
-  extends Transformer[DenseVector[Double],DenseVector[Double]] with Logging {
+case class GaussianMixtureModel(
+    means: DenseMatrix[Double],
+    variances: DenseMatrix[Double],
+    weights: DenseVector[Double],
+    weightThreshold: Double = 1e-4)
+  extends Transformer[DenseVector[Double],DenseVector[Double]] {
+
+  private val gmmMeans = means.t
```

Comment:
Agree with your logic - but we should have a declared convention. What's tricky here is that I *think* that part of the reason we did examples in columns here is because the SIFTs are column-oriented (that is, examples in each column) and therefore so are the PCAs and the GMMs. I am actually inclined to say we should just address this now, but it could have lots of downstream impacts (including on performance) so we should be careful and benchmark before we do too much.